src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -197,17 +197,30 @@
   end
 
   describe 'GET #subprojects' do
-    before do
-      apache_project
-      @project = create(:project, name: 'Apache:Apache2')
-      create(:project, name: 'Apache:Apache2:TestSubproject')
-      create(:project, name: 'Apache:Apache2:TestSubproject2')
-      another_project
-      get :subprojects, params: { project: @project }
+    let!(:apache2) { create(:project, name: 'Apache:Apache2') }
+    let!(:apache2_subproject1) { create(:project, name: 'Apache:Apache2:TestSubproject') }
+    let!(:apache2_subproject2) { create(:project, name: 'Apache:Apache2:TestSubproject2') }
+    let!(:apache2_subprojects) {[apache2_subproject1, apache2_subproject2]}
+
+    context 'subprojects' do
+      before do
+        get :subprojects, params: { project: apache2 }
+      end
+
+      it { expect(assigns(:subprojects)).to match_array(apache2_subprojects) }
+      it { expect(assigns(:parentprojects)).to match_array(apache2.ancestors) }
```

Comment:
actually `apache2.ancestors` should contain `apache_project`, but I removed that by mistake